### PR TITLE
Make classy constraints

### PR DESCRIPTION
**Pull request by mfine**
Move to @JoshuaGross's `makeClassyConstraints`, kill lens boilerplate! Also add a `Makefile`.

Should we call this Lens module something like `Network.Skylark.Core.Lens.TH` instead?

/cc @JoshuaGross @mookerji

### CODE REVIEW

**Comment by JoshuaGross** [comment]
I'm fine with renaming to whatever but I would like to open-source this sooner, rather than later, and putting it in `Network.Skylark.Core` doesn't really make sense long term. But `x.Lens.TH.ClassConstraints` might make sense.

I should have another macro ready tomorrow or the next day, so don't get too attached to using this yet!

**Comment by JoshuaGross** [comment]
This looks great!

File: src/Network/Skylark/Core/Lens/TH.hs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+module Network.Skylark.Core.Lens.TH
+  ( makeClassyConstraints
+  ) where
+
+import Control.Lens
+import Language.Haskell.TH
+import Network.Skylark.Core.Prelude
+
+makeClassyConstraints :: Name -> [Name] -> DecsQ
```
Comment (mookerji) [inline src/Network/Skylark/Core/Lens/TH.hs]: Should note that this doesn't implement the instances for the other classes (like `HasEnv`, etc. below).

**Comment by mfine** [comment]
Moved to TH. This is looking great - working well in the submodules, too! Bringing in.

**Pull request merged**